File: Daily Test/Java/Values - Group By Base 5.11.2025.java
import java.util.*;

public class Main {
    static int baseFromStr(String s) {
        char max = '0';
        for (char c : s.toCharArray())
            if (c > max) max = c;
        return (max - '0') + 1;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String[] nums = new String[n];
        int[] d = new int[20];
        boolean[] used = new boolean[20];
        for (int i = 0; i < n; i++) {
            nums[i] = sc.next();
            int base = baseFromStr(nums[i]);
            d[base] += Integer.parseInt(nums[i], base);
            used[base] = true;
        }
        for (int i = 2; i < 20; i++)
            if (used[i])
                System.out.print(d[i] + " ");
        System.out.println();
    }
}

File: Daily Test/C++/Values - Group By Base 5.11.2025.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cstring>
using namespace std;

int base_from_str(const string &s) {
    char max_digit = '0';
    for (char ch : s)
        if (ch > max_digit) max_digit = ch;
    return (max_digit - '0') + 1;
}

int main() {
    int n; cin >> n;
    vector<string> nums(n);
    vector<int> d(20, 0);
    vector<int> used(20, 0);
    for (int i = 0; i < n; ++i) {
        cin >> nums[i];
        int base = base_from_str(nums[i]);
        d[base] += stoi(nums[i], nullptr, base);
        used[base] = 1;
    }
    for (int i = 2; i < 20; ++i)
        if (used[i]) cout << d[i] << " ";
    cout << endl;
}